org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/quickfix/AbstractDeclarativeIdeQuickfixProvider.java

```diff
@@ -49,7 +58,20 @@ private boolean getFixMethodPredicate(Method input, String issueCode) {
 		return false;
 	}
 
-	private List<DiagnosticResolution> getResolutions(Options options, Iterable<Method> fixMethods) {
+	private DiagnosticResolution getResolution(Options options, Iterable<Method> fixMethods, int id) {
+		DiagnosticResolutionAcceptor issueResolutionAcceptor = issueResolutionAcceptorProvider.get();
+		List<Method> methods = Lists.newArrayList(fixMethods);
+		Method method = methods.get(id);
+		try {
+			// will throw if this is not a public method, but it should be
+			method.invoke(this, issueResolutionAcceptor);
+		} catch (Exception e) {
+			LOG.error("Error executing fix method", e);
+		}
+		return issueResolutionAcceptor.getDiagnosticResolutions(options).get(0);
```

Comment:
A single fix method might contribute multiple resolutions. The hardcoded access to the first item is probably bogus.